src/BenchmarksClient/Startup.cs

```diff
@@ -209,13 +217,58 @@ private static Process StartProcess(ClientJob job)
             process.Exited += (_, __) =>
             {
                 double rps = -1;
-                var match = Regex.Match(job.Output, @"Requests/sec:\s*([\d.]*)");
-                if (match.Success && match.Groups.Count == 2)
+                var rpsMatch = Regex.Match(job.Output, @"Requests/sec:\s*([\d\.]*)");
+                if (rpsMatch.Success && rpsMatch.Groups.Count == 2)
                 {
-                    double.TryParse(match.Groups[1].Value, out rps);
+                    double.TryParse(rpsMatch.Groups[1].Value, out rps);
                 }
                 job.RequestsPerSecond = rps;
 
+                double average = -1;
+                var latencyMatch = Regex.Match(job.Output, @"Latency\s*([\d\.]*)");
+                if (latencyMatch.Success && latencyMatch.Groups.Count == 2)
+                {
+                    double.TryParse(latencyMatch.Groups[1].Value, out average);
+                }
+                job.Latency.Average = average;
+
+                // Start Latency Distribution pattent matching after a specific index as 
+                // previous results could render a .75% result for instance
+                var latencyDistributionIndex = job.Output.IndexOf("Latency Distribution");
+
+                double p50 = -1;
+                var p50Match = new Regex(@"50%\s*([\d\.]*)").Match(job.Output, latencyDistributionIndex);
+                if (p50Match.Success && p50Match.Groups.Count == 2)
+                {
+                    double.TryParse(p50Match.Groups[1].Value, out p50);
```

Comment:
If `TryParse()` fails, the client will silently set latency to `-1`.  If parsing should never fail, I think it would be better for the client to throw and shutdown.  This should prevent the driver from writing invalid latency data to SQL.

Reply:
And yes, we should do the same for RPS.  This was a bug before your changes.